x/staking/keeper/msg_server.go

```diff
@@ -74,6 +74,13 @@ func (k msgServer) CreateValidator(goCtx context.Context, msg *types.MsgCreateVa
 	if err != nil {
 		return nil, err
 	}
+
+	if ctx.BlockHeight() >= 712000 {
```

Comment:
You cannot generically apply what Osmosis did to the SDK. They were only able to do this in a non-state-breaking manner because they used a height-gated switch (as seen here). But you cannot generally apply this to all chains.

You could make a public `var` that an app must set, but I honestly don't think that is worth it.